mongo/datadog_checks/mongo/mongo.py

```diff
@@ -319,6 +328,60 @@ def _authenticate(self, database):
 
         return authenticated
 
+    def _parse_config(self):
+        """
+        See https://docs.mongodb.com/manual/reference/connection-string/
+        """
+        hosts = self.instance.get('hosts', [])
+        if not hosts:
+            raise ConfigurationError('No `hosts` specified')
+
+        connection_scheme = self.instance.get('connection_scheme', 'mongodb')
+        username = self.instance.get('username')
+        password = self.instance.get('password')
+        database = self.instance.get('database')
+        options = self.instance.get('options', {})
+
+        # Construct the connection string
+        server = '{}://'.format(connection_scheme)
+
+        if username and password:
+            server += '{}:{}@'.format(username, password)
+
+        server += ','.join(hosts)
+
+        if database:
+            server += '/{}'.format(database)
+
+        if options:
+            if not database:
+                server += '/'
+
+            server += '?{}'.format('&'.join('{}={}'.format(key, value) for key, value in options.items()))
+
+        # Satisfy requirements of _parse_uri
+        auth_source = None
+        for key, value in options.items():
+            if key.lower() == 'authsource':
+                auth_source = value
+                break
+
+        nodelist = []
+        for host in hosts:
+            if ':' in host:
+                hostname, port = host.rsplit(':', 1)
+                nodelist.append((hostname, int(port)))
+            else:
+                nodelist.append((host, 27017))
+
+        clean_server_name = server.replace(password, '*' * 5) if password else server
+
+        if username and self.ssl_params:
```

Comment:
Just for information, do you know why we hide the username when ssl is active ? What's the reason ?

Reply:
I have no idea and was about to ask you!

Reply:
Same here, let's keep it like this then. The effect of that is negligible.